lib/config/migration.ts

```diff
@@ -253,6 +253,18 @@ export function migrateConfig(
             );
           }
         }
+      } else if (key === 'unpublishSafe') {
+        isMigrated = true;
+        if (val === true) {
+          migratedConfig.extends = migratedConfig.extends || [];
+          if (is.string(migratedConfig.extends)) {
```

Comment:
There is potential problem here, when single string requires migration but we already checked `key === 'extends' && is.array<string>(val)`. We need either coerce string to array also for previous `else if` branch, or don't coerce at all.

Reply:
`extends` should be an array, but I think we allow/migrate a single string too. So ideally this check wouldn't be required but it's fairly easy to add it. Can you give an example of a config that wouldn't migrate correctly?

Reply:
Given fields are processed in this order:
```json
{
  "extends": "config:application",
  "unpublishSafe": true,
}
```
It would migrate `'config:application'` to `['config:application', 'npm:unpublishSafe']`, but not to `['config:js-app', 'npm:unpublishSafe']`. But yes, it's easy to fix it.

Reply:
OK thanks, please fix the `extends` migration in that case